auth/src/main/java/com/firebase/ui/auth/ui/credentials/CredentialSaveActivity.java

```diff
@@ -0,0 +1,122 @@
+package com.firebase.ui.auth.ui.credentials;
+
+import android.arch.lifecycle.Observer;
+import android.arch.lifecycle.ViewModelProviders;
+import android.content.Context;
+import android.content.Intent;
+import android.content.IntentSender;
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.util.Log;
+
+import com.firebase.ui.auth.IdpResponse;
+import com.firebase.ui.auth.data.model.FlowParameters;
+import com.firebase.ui.auth.data.model.Resource;
+import com.firebase.ui.auth.ui.HelperActivityBase;
+import com.firebase.ui.auth.util.CredentialsUtil;
+import com.firebase.ui.auth.util.ExtraConstants;
+import com.firebase.ui.auth.util.data.ProviderUtils;
+import com.firebase.ui.auth.viewmodel.PendingResolution;
+import com.firebase.ui.auth.viewmodel.ResolutionCodes;
+import com.firebase.ui.auth.viewmodel.smartlock.SmartLockHandler;
+import com.google.android.gms.auth.api.credentials.Credential;
+import com.google.firebase.auth.FirebaseUser;
+
+/**
+ * Invisible Activity used for saving credentials to SmartLock.
+ */
+public class CredentialSaveActivity extends HelperActivityBase {
+
+    private static final String TAG = "SmartlockSave";
+
+    private SmartLockHandler mHandler;
+    private IdpResponse mIdpResponse;
+
+    @NonNull
+    public static Intent createIntent(Context context,
+                                      FlowParameters flowParams,
+                                      FirebaseUser firebaseUser,
+                                      @Nullable String password,
+                                      IdpResponse response) {
+
+        String accountType = ProviderUtils.idpResponseToAccountType(response);
+        Credential credential = CredentialsUtil.buildCredential(firebaseUser, password, accountType);
+
+        return HelperActivityBase.createBaseIntent(context, CredentialSaveActivity.class, flowParams)
+                .putExtra(ExtraConstants.EXTRA_CREDENTIAL, credential)
+                .putExtra(ExtraConstants.EXTRA_IDP_RESPONSE, response);
+    }
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        mHandler = ViewModelProviders.of(this).get(SmartLockHandler.class);
+        mHandler.init(getFlowParams());
+
+        Credential credential = getIntent().getParcelableExtra(ExtraConstants.EXTRA_CREDENTIAL);
+        mIdpResponse = getIntent().getParcelableExtra(ExtraConstants.EXTRA_IDP_RESPONSE);
+
+        mHandler.saveCredentials(credential);
```

Comment:
This should be guarded by a `savedInstanceState == null` check. (Or should we make the activity resilient to config changes with `android:configChanges="keyboard|keyboardHidden|screenLayout|screenSize|orientation"`?)